# RNA-seq

- –†–µ–∑—É–ª—å—Ç–∞—Ç
    
    

# –í–≤–µ–¥–µ–Ω–∏–µ

–¶–µ–ª—å—é –¥–∞–Ω–Ω–æ–≥–æ –∑–∞–¥–∞–Ω–∏—è —è–≤–ª—è–µ—Ç—Å—è —Å—Ä–∞–≤–Ω–µ–Ω–∏–µ RNA-seq –¥–∞–Ω–Ω—ã—Ö –ø–µ—Ä–µ–ø—Ä–æ–≥—Ä–∞–º–º–∏—Ä–æ–≤–∞–Ω–Ω—ã—Ö –∏ –Ω–µ–ø–µ—Ä–µ–ø—Ä–æ–≥—Ä–∞–º–º–∏—Ä–æ–≤–∞–Ω–Ω—ã—Ö (–∫–æ–Ω—Ç—Ä–æ–ª—å–Ω—ã—Ö) –º—ã—à–∏–Ω—ã—Ö —ç–º–±—Ä–∏–æ–Ω–∞–ª—å–Ω—ã—Ö —Ñ–∏–±—Ä–æ–±–ª–∞—Å—Ç–æ–≤ (MEFs) –∏ –Ω–∞—Ö–æ–∂–¥–µ–Ω–∏–µ –≥–µ–Ω–æ–≤, –∫–æ—Ç–æ—Ä—ã–µ –Ω–∞–∏–±–æ–ª–µ–µ —Å–∏–ª—å–Ω–æ –∏–∑–º–µ–Ω—è—é—Ç —Å–≤–æ—é —ç–∫—Å–ø—Ä–µ—Å—Å–∏—é –≤ —ç—Ç–æ–º –ø—Ä–æ—Ü–µ—Å—Å–µ. –í –∫–∞—á–µ—Å—Ç–≤–µ –¥–æ–ø–æ–ª–Ω–∏—Ç–µ–ª—å–Ω–æ–≥–æ –∑–∞–¥–∞–Ω–∏—è –ø—Ä–µ–¥–ª–∞–≥–∞–µ—Ç—Å—è –∏–∑—É—á–∏—Ç—å –æ—Å–Ω–æ–≤–Ω—ã–µ. 

# –ü–æ–¥–≥–æ—Ç–æ–≤–∏–º —Ä–∞–±–æ—á–µ–µ –ø—Ä–æ—Å—Ç—Ä–∞–Ω—Å—Ç–≤–æ

–Ø —Ä–∞–±–æ—Ç–∞—é –Ω–∞ –∫–æ–º–ø—É—Ö—Ç–µ—Ä–µ —Å –æ–ø–µ—Ä–∞—Ü–∏–æ–Ω–Ω–æ–π —Å–∏—Å—Ç–µ–º–æ–π Windows. –ü–æ—ç—Ç–æ–º—É –ø–µ—Ä–µ–¥ –Ω–∞—á–∞–ª–æ–º —Ä–∞–±–æ—Ç—ã –±—ã–ª–∏ —É—Å—Ç–∞–Ω–æ–≤–ª–µ–Ω—ã `WSL` –∏ `Ubuntu 22.04.3`. –¢–∞–∫ –∂–µ –¥–æ–ø–æ–ª–Ω–∏—Ç–µ–ª—å–Ω–æ –±—ã–ª–∞ —É—Å—Ç–∞–Ω–æ–≤–ª–µ–Ω–∞ `Anaconda3-2023.09-0-Linux-x86_64`

–ü–µ—Ä–µ–¥–∞ –Ω–∞—á–∞–ª–æ–º –ø–æ–¥–≥–æ—Ç–æ–≤–∏–º —Ä–∞–±–æ—á–µ–µ –ø—Ä–æ—Å—Ç—Ä–∞–Ω—Å—Ç–≤–æ, –∞ –∏–º–µ–Ω–Ω–æ —Å–æ–∑–¥–∞–¥–∏–º —Ä–∞–±–æ—á—É—é –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏—é –∏ –≤—Å–µ –Ω–µ–æ–±—Ö–æ–¥–∏–º—ã–µ –≤–ª–æ–∂–µ–Ω–Ω—ã–µ —Ñ–∞–π–ª—ã.

```bash
# –°–æ–∑–¥–∞–¥–∏–º —Ä–∞–±–æ—á—É—é –¥–∏—Ä—Ä–µ–∫—Ç–æ—Ä–∏—é –∏ –ø–µ—Ä–µ–º–µ—Å—Ç–∏–º—Å—è –≤ –Ω–µ—ë
$ mkdir -p omics_hw/rnaseq
$ cd omics_hw/rnaseq

# C–æ–∑–¥–∞–¥–∏–º –Ω–µ–æ–±—Ö–æ–¥–∏–º—ã–µ –¥–∏—Ä—Ä–µ–∫—Ç–æ—Ä–∏–∏
$ mkdir fastq_files # –¥–∏—Ä—Ä–µ–∫—Ç–æ—Ä–∏—è —Å —á–∏—Å—Ç—ã–º–∏ .fastq —Ñ–∞–π–ª—ã
$ mkdir scripts # –¥–∏—Ä—Ä–µ–∫—Ç–æ—Ä–∏—è —Å–æ —Å–∫—Ä–∏–ø—Ç–∞–º–∏ —Å –ø–æ–º–æ—â—å—é –∫–æ—Ç–æ—Ä—ã—Ö –º—ã –±—É–¥–µ–º –æ–±—Ä–∞–±–∞—Ç—ã–≤–∞—Ç—å –¥–∞–Ω–Ω—ã–µ
$ mkdir references # –¥–∏—Ä—Ä–µ–∫—Ç–æ—Ä–∏—è c –≥–µ–Ω–æ–º–Ω—ã–º —Ä–µ—Ñ–µ—Ä–µ–Ω—Å–æ–º, –∞–Ω–Ω–æ—Ç–∞—Ü–∏–µ–π –∏ –∏–Ω–¥–µ–∫—Å–∞–º–∏ –¥–ª—è –≥–µ–Ω–æ–º–∞
$ mkdir qc # –¥–∏—Ä—Ä–µ–∫—Ç–æ—Ä–∏—è —Å –æ—Ç—á—ë—Ç–∞–º–∏ –æ –∫–∞—á–µ—Å—Ç–≤–µ –Ω–∞—à–∏—Ö –¥–∞–Ω–Ω—ã—Ö
```

# –°–∫–∞—á–∏–≤–∞–Ω–∏–µ —Ñ–∞–π–ª–æ–≤

–°–∫–∞—á–∞–µ–º –¥–∞–Ω–Ω—ã–µ PRJNA836496 –ø—Ä–∏ –ø–æ–º–æ—â–∏ [SRA Explorer](https://sra-explorer.info/#). –î–ª—è —ç—Ç–æ–≥–æ —Å–æ–∑–¥–∞–¥–∏–º —Å–∫—Ä–∏–ø—Ç `load_fastq.sh` –≤ –¥–∏—Ä—Ä–µ–∫—Ç–æ—Ä–∏–∏ `scripts`. –î–æ–±–∞–≤–∏–º `$1` –ø–µ—Ä–µ–¥ –Ω–∞–∑–≤–∞–Ω–∏–µ–º —Ñ–∞–π–ª–∞ —á—Ç–æ–±—ã —É–∫–∞–∑–∞—Ç—å –Ω–µ–æ–±—Ö–æ–¥–∏–º—É—é –Ω–∞–º –¥–∏—Ä–µ–∫—Ç–æ—Ä–∏—é.

```bash
#!/usr/bin/env bash
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR341/009/SRR3414629/SRR3414629.fastq.gz -o $1/SRR3414629_GSM2130680_MEF_OSKM_induced_Day7_Scrambled_shRNA_replicate1_Mus_musculus_RNA-Seq.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR341/000/SRR3414630/SRR3414630.fastq.gz -o $1/SRR3414630_GSM2130681_MEF_OSKM_induced_Day7_Scrambled_shRNA_replicate2_Mus_musculus_RNA-Seq.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR341/001/SRR3414631/SRR3414631.fastq.gz -o $1/SRR3414631_GSM2130682_MEF_OSKM_induced_Day7_Scrambled_shRNA_replicate3_Mus_musculus_RNA-Seq.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR341/006/SRR3414636/SRR3414636.fastq.gz -o $1/SRR3414636_GSM2130687_MEF_Day7_Scrambled_shRNA_replicate2_Mus_musculus_RNA-Seq.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR341/005/SRR3414635/SRR3414635.fastq.gz -o $1/SRR3414635_GSM2130686_MEF_Day7_Scrambled_shRNA_replicate1_Mus_musculus_RNA-Seq.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR341/007/SRR3414637/SRR3414637.fastq.gz -o $1/SRR3414637_GSM2130688_MEF_Day7_Scrambled_shRNA_replicate3_Mus_musculus_RNA-Seq.fastq.gz
```

–°–∫–∞—á–∞–µ–º `.fastq` —Ñ–∞–π–ª—ã –¥–ª—è –Ω—É–∂–Ω—ã—Ö –Ω–∞–º –æ–±—Ä–∞–∑—Ü–æ–≤

```bash
$ bash scripts/load_fastq.sh fastq_files
```

–ü–µ—Ä–µ–∏–º–µ–Ω—É–µ–º —Ñ–∞–π–ª—ã —á—Ç–æ–±—ã —Å –Ω–∏–º–∏ –±—ã–ª–æ –ø—Ä–æ—â–µ —Ä–∞–±–æ—Ç–∞—Ç—å. –î–ª—è —ç—Ç–æ–≥–æ —Å–¥–µ–ª–∞–µ–º –Ω–µ–±–æ–ª—å—à–æ–π —Å–∫—Ä–∏–ø—Ç –Ω–∞ `python` 

```python
import os
import glob

dir_name = 'fastq_files'
for filename in glob.glob(f'{dir_name}/*'):
    sra_code = filename.split('/')[1].split('_')[0]
    new_name = f'{dir_name}/{sra_code}.fastq.gz'
    os.rename(filename, new_name)
    print(sra_code)

```

–ó–∞–ø—É—Å—Ç–∏–º —Å–∫—Ä–∏–ø—Ç

```bash
$ python3 scripts/rename_fastq.py
```

# QC –¥–ª—è —Å—ã—Ä—ã—Ö –æ–±—Ä–∞–∑—Ü–æ–≤

–°–æ–∑–¥–∞–¥–∏–º –∞–Ω–∞–∫–æ–Ω–¥–æ–≤—Å–∫–æ–µ –æ–∫—Ä—É–∂–µ–Ω–∏–µ –∏ —É—Å—Ç–∞–Ω–æ–≤–∏–º  `fastqc`, `multiqc`

```bash
$ conda create -n preprocess -c conda-forge -c bioconda fastqc multiqc
$ conda activate preprocess
```

–¢–µ–ø–µ—Ä—å —Å–¥–µ–ª–∞–µ–º —Ñ–∞–π–ª —Å –∑–∞–ø—É—Å–∫–æ–º –∞–Ω–∞–ª–∏–∑–∞ –∫–∞—á–µ—Å—Ç–≤–∞ –ø—Ä–æ—á—Ç–µ–Ω–∏–π

```bash
#!/bin/bash
### Run QC
fastqc -o qc -t 8 fastq_files/*.fastq.gz

### Run MultiQC
multiqc qc --outdir qc --filename multiqc.html --title "Read quality" --force
```

–ó–∞–ø—É—Å—Ç–∏–º —Å–∫—Ä–∏–ø—Ç

```bash
$ bash scripts/qs_script.sh
```

- *–†–µ–∑—É–ª—å—Ç–∞—Ç—ã*
    
    [multiqc.html](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/multiqc.html)
    

# –ö–∞—Ä—Ç–∏—Ä–æ–≤–∞–Ω–∏–µ

## –°–∫–∞—á–∞–µ–º —Ñ–∞–π–ª—ã —Å —Ä–µ—Ñ–µ—Ä–µ–Ω—Å–æ–º

–°–æ—Å—Ç–∞–≤–∏–º —Å–∫—Ä–∏–ø—Ç –¥–ª—è —Å–∫–∞—á–∏–≤–∞–Ω–∏—è

```bash
#!/bin/bash
# —Å–∫–∞—á–∞–µ–º —Ä–µ—Ñ–µ—Ä–µ–Ω—Å–Ω—ã–π –≥–µ–Ω–æ–º –¥–ª—è hisat2 —Å ENSEMBL
wget https://ftp.ensembl.org/pub/release-111/fasta/mus_musculus/dna/Mus_musculus.GRCm39.dna_sm.primary_assembly.fa.gz
gzip -d Mus_musculus.GRCm39.dna_sm.primary_assembly.fa.gz
mv Mus_musculus.GRCm39.dna_sm.primary_assembly.fa references/genome.fa

# —Å–∫–∞—á–∞–µ–º –≥–µ–Ω–æ–º–Ω—É—é –∞–Ω–Ω–æ—Ç–∞—Ü–∏—é —Å ENSEMBL
wget https://ftp.ensembl.org/pub/release-111/gtf/mus_musculus/Mus_musculus.GRCm39.111.gtf.gz
gzip -d Mus_musculus.GRCm39.111.gtf.gz
mv Mus_musculus.GRCm39.111.gtf references/genome.gtf

# —Å–∫–∞—á–∞–µ–º —Ä–µ—Ñ–µ—Ä–µ–Ω—Å–Ω—ã–π —Ç—Ä–∞–Ω—Å–∫—Ä–∏–ø—Ç–æ–º –¥–ª—è kallisto —Å ENSEMBL
wget https://ftp.ensembl.org/pub/release-111/fasta/mus_musculus/cdna/Mus_musculus.GRCm39.cdna.all.fa.gz
gzip -d Mus_musculus.GRCm39.cdna.all.fa.gz
mv Mus_musculus.GRCm39.cdna.all.fa references/transcriptome.fa
```

–ó–∞–ø—É—Å—Ç–∏–º —Å–∫—Ä–∏–ø—Ç

```bash
$ bash scripts/load_reference.sh
```

## Hisat2

–°–∫–∞—á–∞–µ–º `hisat2` –≤ –Ω–∞—à–µ –æ–∫—Ä—É–∂–µ–Ω–∏–µ

```bash
$ conda install hisat2 -c bioconda
```

–ü—Ä–æ–∏–Ω–¥–µ–∫—Å–∏—Ä—É–µ–º –≥–µ–Ω–æ–º

```bash
$ mkdir -p references/hisatIndex # —Å–æ–∑–¥–∞–¥–∏–º –ø–∞–ø–∫—É –¥–ª—è –∏–Ω–¥–µ–∫—Å–æ–≤
$ hisat2-build -p 8 references/genome.fa references/hisatIndex/musgrcm39Index # –∑–∞–ø—É—Å—Ç–∏–º –∏–Ω–¥–µ–∫—Å–∏—Ä–æ–≤–∞–Ω–∏–µ
```

–¢–µ–ø–µ—Ä—å –∑–∞–ø—É—Å—Ç–∏–º –∫–∞—Ä—Ç–∏—Ä–æ–≤–∞–Ω–∏–µ –æ–±—Ä–∞–∑—Ü–æ–≤

```bash
#!/usr/bin/env bash
# make directories for quantified samples
mkdir -p hisatOutput

# hisat alignment
for infile in fastq_files/*.fastq.gz
do
  base=$(basename -a -s .fastq.gz ${infile})
  mkdir -p hisatOutput/${base}
  hisat2 -p 8 \
  --phred33 \
  -x references/hisatIndex/musgrcm39Index \
  -U $infile \
  -S hisatOutput/${base}/alignment.sam \
  --summary-file hisatOutput/${base}/summary.txt
done
```

–ó–∞–ø—É—Å—Ç–∏–º —Å–∫—Ä–∏–ø—Ç

```bash
$ bash scripts/run_hisat_alignment.sh
```

–î–ª—è –¥–∞–ª—å–Ω–µ–π—à–µ–π —Ä–∞–±–æ—Ç—ã –Ω–∞–º –Ω—É–∂–Ω–æ –ø–µ—Ä–µ–≤–µ—Å—Ç–∏ —Ñ–∞–π–ª—ã –∏–∑ —Ñ–æ—Ä–º–∞—Ç–∞ `.sam` –≤ —Ñ–æ—Ä–º–∞—Ç `.bam`, –æ—Ç—Å–æ—Ä—Ç–∏—Ä–æ–≤–∞—Ç—å –∏—Ö –∏ –ø—Ä–æ–∏–Ω–¥–µ–∫—Å–∏—Ä–æ–≤–∞—Ç—å. –ß—Ç–æ–±—ã —ç—Ç–æ —Å–¥–µ–ª–∞—Ç—å –≤–æ—Å–ø–æ–ª—å–∑—É–µ–º—Å—è `samtools`.

```bash
$ conda install samtools
```

<aside>
üñäÔ∏è –ù–µ –±—É–¥–µ–º –≤—ã–±—Ä–∞—Å—ã–≤–∞—Ç—å —Ä–∏–¥—ã, –∫–æ—Ç–æ—Ä—ã–µ –æ—Ç–∫–∞—Ä—Ç–∏—Ä–æ–≤–∞–ª–∏—Å—å –Ω–µ—É–Ω–∏–∫–∞–ª—å–Ω–æ, —Ç.–∫. `htseq` —Å–∞–º–æ—Å—Ç–æ—è—Ç–µ–ª—å–Ω–æ –Ω–µ –±—É–¥–µ—Ç —É—á–∏—Ç—ã–≤–∞—Ç—å –∏—Ö –ø—Ä–∏ –ø–æ–¥—Å—á—ë—Ç–µ —ç–∫—Å–ø—Ä–µ—Å—Å–∏–∏ (—Ç–æ–ª—å–∫–æ –µ—Å–ª–∏ –µ–≥–æ –Ω–µ –ø–æ–ø—Ä–æ—Å–∏—Ç—å –æ–±—Ä–∞—Ç–Ω–æ–µ)

</aside>

- *P.S. –∫–∞–∫* `htseq` *—Å—á–∏—Ç–∞–µ—Ç —Ä–∏–¥—ã*
    
    –î–æ–∫—É–º–µ–Ω—Ç–∞—Ü–∏—è: [—Å—Å—ã–ª–∫–∞](https://htseq.readthedocs.io/en/latest/htseqcount.html)
    
    ![Untitled](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled.png)
    

–ù–∞–ø–∏—à–µ–º —Å–∫—Ä–∏–ø—Ç. –ü–æ–ª–æ–∂–∏–º –æ—Ç—Å–æ—Ä—Ç–∏—Ä–æ–≤–∞–Ω—ã–µ `.bam` —Ñ–∞–π–ª—ã –∏ –∏—Ö –∏–Ω–¥–µ–∫—Å—ã –≤ –æ—Ç–¥–µ–ª—å–Ω—É—é –ø–∞–ø–∫—É —Ñ–∞–π–ª—ã –≤ –æ—Ç–¥–µ–ª—å–Ω—É—é –ø–∞–ø–∫—É.

```bash
#!/bin/bash

mkdir -p sortedbam

for infile in hisatOutput/*
do
  sample=$(basename -a ${infile})
  samtools view -@4 -b hisatOutput/${sample}/alignment.sam > hisatOutput/${sample}/alignment.bam
  samtools sort -@4 hisatOutput/${sample}/alignment.bam -o sortedbam/${sample}_sorted.bam
  samtools index -@4 sortedbam/${sample}_sorted.bam
done
```

–ó–∞–ø—É—Å—Ç–∏–º —Å–∫—Ä–∏–ø—Ç

```bash
$ bash scripts/sambam_sort_index.sh
```

## Kallisto

–í –∫–∞—á–µ—Å—Ç–≤–µ –¥–æ–ø–æ–ª–Ω–∏—Ç–µ–ª—å–Ω–æ–≥–æ –∑–∞–¥–∞–Ω–∏—è —Å–¥–µ–ª–∞–µ–º –∫–≤–∞–Ω—Ç–∏—Ñ–∏–∫–∞—Ü–∏—é —ç–∫—Å–ø—Ä–µ—Å—Å–∏–∏ –ø—Ä–∏ –ø–æ–º–æ—â–∏ `kallisto`, –∫–æ—Ç–æ—Ä—ã–π –≤—ã–ø–æ–ª–Ω—è–µ—Ç –ø—Å–µ–≤–¥–æ–≤—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏–µ.

–°–∫–∞—á–∞–µ–º [kallisto](http://pachterlab.github.io/kallisto/) –≤ –Ω–∞—à–µ –æ–∫—Ä—É–∂–µ–Ω–∏–µ. –Ø –∑–¥–µ—Å—å —Å–∫–∞—á–∏–≤–∞—é –≤–µ—Ä—Å–∏—é 0.48.0 –ø–æ—Ç–æ–º—É-—á—Ç–æ –±–æ–ª–µ–µ –Ω–æ–≤—ã–µ –≤–µ—Ä—Å–∏–∏ —É –º–µ–Ω—è –Ω–µ —Ä–∞–±–æ—Ç–∞—é—Ç. –°—Ç–∞–ª–∫–∏–≤–∞—é—Å—å —Å–æ —Å–ª–µ–¥—É—é—â–µ–π –æ—à–∏–±–∫–æ–π https://github.com/pachterlab/kallisto/issues/399

```bash
$ conda install -c conda-forge -c bioconda kallisto==0.48.0
```

–¢–µ–ø–µ—Ä—å –Ω–∞–º –Ω—É–∂–Ω–æ –ø—Ä–æ–∏–Ω–¥–µ–∫—Å–∏—Ä–æ–≤–∞—Ç—å —Ç—Ä–∞–Ω—Å–∫—Ä–∏–ø—Ç–æ–º. 

```bash
$ kallisto index -i references/musGRCm39_kallisto references/transcriptome.fa
```

–¢–µ–ø–µ—Ä—å –Ω–∞–ø–∏—à–µ–º —Å–∫—Ä–∏–ø—Ç –¥–ª—è –∫–≤–∞–Ω—Ç–∏—Ñ–∏–∫–∞—Ü–∏–∏ –Ω–∞—à–∏—Ö —Ä–∏–¥–æ–≤ –ø—Ä–∏ –ø–æ–º–æ—â–∏ kallisto. –Ø –Ω–µ –Ω–∞—à—ë–ª –∏–Ω—Ñ–æ—Ä–º–∞—Ü–∏—é –æ –¥–ª–∏–Ω–µ —Ñ—Ä–∞–≥–º–µ–Ω—Ç–æ–≤ –≤ –æ–ø–∏—Å–∞–Ω–∏–∏ —ç–∫—Å–ø–µ—Ä–∏–º–µ–Ω—Ç–∞ –Ω–∞ [GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE80550). –ü–æ—ç—Ç–æ–º—É –±—É–¥–µ–º –∑–∞–ø—É—Å–∫–∞—Ç—å —Å–æ —Å—Ç–∞–Ω–¥–∞—Ä—Ç–Ω—ã–º–∏ –ø–∞—Ä–∞–º–µ—Ç—Ä–∞–º–∏. –ë—É–¥–µ–º —Å–æ—Ö—Ä–∞–Ω—è—Ç—å –ª–æ–≥–∏ –≤ –æ—Ç–¥–µ–ª—å–Ω—ã–π —Ñ–∞–π–ª —á—Ç–æ–±—ã –≤–∫–ª—é–∏—Ç—å –∏—Ö –≤ –æ—Ç—á—ë—Ç `multiqc`

```bash
#!/bin/bash
# make directories for quantified samples
mkdir -p kallistoOutput

# kallisto quantification step
for infile in fastq_files/*.fastq.gz
do
  sample=$(basename -a -s .fastq.gz ${infile})
  echo ${sample}
  kallisto quant -i references/musGRCm39_kallisto -o kallistoOutput/${sample} --single -l 180 -s 20.0 -t 8 fastq_files/${sample}.fastq.gz &> kallistoOutput/${sample}/stdout.log
done
```

<aside>
üñäÔ∏è For single end reads, you¬†**must**¬†supply the length and standard deviation of the fragment length (not the read length) ([quick note about the difference between fragment length and read length](https://www.ecseq.com/support/ngs/why-do-the-reads-all-have-the-same-length-when-sequencing-differently-sized-fragments)). The exact information can be obtained by Bioanalyzer/Fragment Analyzer results on the prepared RNA-seq libraries. If you do not have this information, using fragment length = 180 and sd = 20 (like Kallisto manual page) could be a good start.

</aside>

–ó–∞–ø—É—Å—Ç–∏–º —Å–∫—Ä–∏–ø—Ç

```python
$ bash scripts/kallisto_quantify.sh
```

- –†–µ–∑—É–ª—å—Ç–∞—Ç—ã
    
    [multiqcfinal.html](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/multiqcfinal.html)
    

# Alignment QC

–¢–µ–ø–µ—Ä—å –¥–∞–≤–∞–π—Ç–µ –ø—Ä–æ–≤–µ—Ä–∏–º –∫–∞—á–µ—Å—Ç–≤–æ –≤—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏—è. –°–æ–∑–¥–∞–¥–∏–º –Ω–æ–≤–æ–µ –æ–∫—Ä—É–∂–µ–Ω–∏–µ –¥–ª—è RSeQC

```bash
$ conda create -n rseqc
$ conda activate rseqc
$ conda install -c bioconda rseqc
```

–î–ª—è –∑–∞–ø—É—Å–∫–∞ —Å–∞–º–æ–π –∏–Ω—Ç–µ—Ä–µ—Å–Ω–æ–π —Ñ—É–Ω–∫—Ü–∏–∏ `geneBody_coverage` –Ω–∞–º –ø–æ–Ω–∞–¥–æ–±–∏—Ç—Å—è —Å–¥–µ–ª–∞—Ç—å –∫–∞—Å—Ç–æ–º–Ω—ã–π `.bed` —Ñ–∞–π–ª –Ω–∞ –æ—Å–Ω–æ–≤–µ –Ω–∞—à–µ–≥–æ `.gtf` —Ñ–∞–π–ª–∞. –ë–æ–ª–µ–µ —Ç–æ–≥–æ, `geneBody_coverage` —Ä–∞–±–æ—Ç–∞–µ—Ç –æ—á–µ–Ω—å –¥–æ–ª–≥–æ (—Ç.–∫. –Ω–∞–ø–∏—Å–∞–Ω–∞ –Ω–∞ `python`), –ø–æ—ç—Ç–æ–º—É –º—ã –±—ã —Ö–æ—Ç–µ–ª–∏ –∑–∞–ø—É—Å—Ç–∏—Ç—å –µ—ë –Ω–∞ –Ω–µ–±–æ–ª—å—à–æ–π –ø–æ–¥–≤—ã–±–æ—Ä–∫–µ –≥–µ–Ω–æ–≤. –ë—É–¥–µ–º –∑–∞–ø—É—Å–∫–∞—Ç—å –Ω–∞ –≥–µ–Ω–∞—Ö –¥–æ–º–∞—à–Ω–µ–≥–æ —Ö–æ–∑—è–π—Å—Ç–≤–∞. –î–ª—è —ç—Ç–æ–≥–æ –≤–æ–∑—å–º—ë–º –∏—Ö Refseq –∏–¥–µ–Ω—Ç–∏—Ñ–∏–∫–∞—Ç–æ—Ä—ã –∏–∑ —Ñ–∞–π–ª–∞ ([mm10.HouseKeepingGenes.bed.gz](https://sourceforge.net/projects/rseqc/files/BED/Mouse_Mus_musculus/mm10.HouseKeepingGenes.bed.gz/download)) —Å —Å–∞–π—Ç–∞ –¥–æ–∫—É–º–µ–Ω—Ç–∞—Ü–∏–∏ [RseQC](https://sourceforge.net/projects/rseqc/files/BED/Mouse_Mus_musculus/).

<aside>
üñäÔ∏è –ü–æ-—Ö–æ—Ä–æ—à–µ–º—É –Ω—É–∂–Ω–æ –±—ã–ª–æ –±—ã —Å–¥–µ–ª–∞—Ç—å —Ç–∞–∫ –∂–µ –∫–∞–∫ –∏ –æ–Ω–∏:
1. –í–∑—è—Ç—å –≥–µ–Ω—ã –¥–æ–º–∞—à–Ω–µ–≥–æ —Ö–æ–∑—è–π—Å—Ç–≤–∞ –¥–ª—è —á–µ–ª–æ–≤–µ–∫–∞
2. –ù–∞–π—Ç–∏ –¥–ª—è –Ω–∏—Ö –≥–æ–º–æ–ª–æ–≥–∏ —É –º—ã—à–∏

</aside>

–ù–∞–ø–∏—à–µ–º —Å–∫—Ä–∏–ø—Ç —á—Ç–æ–±—ã –≤—ã—Ç–∞—â–∏—Ç—å –Ω—É–∂–Ω—ã–µ –Ω–∞–º –≥–µ–Ω—ã –∏–∑ `.bed` —Ñ–∞–π–ª–∞, –∫–æ—Ç–æ—Ä—ã–π –º—ã —Å–∫–∞—á–∞–ª–∏, –∞ —Ç–∞–∫ –∂–µ —Å–æ—Å—Ç–∞–≤–∏—Ç—å –Ω–∞—à `.bed` —Ñ–∞–π–ª –Ω–∞ –æ—Å–Ω–æ–≤–µ [Biomart](https://www.ensembl.org/biomart/martview/87c698ebbceb6b8e05466761bcb6fa0c) –∞–Ω–Ω–æ—Ç–∞—Ü–∏–∏. 

```python
import pandas as pd

# load a list of house-keeping genes
hk_df = pd.read_csv('references/mm10.HouseKeepingGene.bed', sep='\t', header=None)
hk_df.columns = ['chr', 'start', 'end', 'name', 'score', 'strand', 'thickStart', 'thickEnd', 'ItemRgb', 'blockCount', 'blockSizes', 'blockStarts']
hk_df = hk_df.set_index('name')
refseq_id = list(hk_df.index.unique())

# load gtf file to get chr positions
gtf  = pd.read_csv("references/genome.gtf", sep='\t', header=None, skiprows=5, usecols=[0, 3, 4, 8], dtype='str')
gtf.columns = ['chr_ind', 'chrstart', 'chrend', 'Gene stable ID']
gtf['Gene stable ID'] = gtf['Gene stable ID'].map(lambda x: x.split(';')[0].replace('gene_id "', '').replace('"', '') if 'transcript_id' not in x else None)
gtf.dropna(inplace=True)

# load biomart annotation to get refseq ids
biomart_df = pd.read_csv("references/biomart_export.txt", sep='\t')
biomart_hk = biomart_df[biomart_df["RefSeq mRNA ID"].isin(refseq_id)]
biomart_hk = biomart_hk.merge(gtf, on='Gene stable ID').set_index('RefSeq mRNA ID')

# create a bed file
merge_col = ['Gene stable ID', 'chrstart', 'chrend', 'chr_ind']
merge_df = hk_df.merge(biomart_hk[merge_col], left_index=True, right_index=True).reset_index()
bed_file = merge_df[['chr_ind', 'chrstart', 'chrend', 'Gene stable ID', 'score', 'strand', 'thickStart', 'thickEnd', 'ItemRgb', 'blockCount', 'blockSizes', 'blockStarts']]
bed_file.to_csv('references/hk_genes_ensembl.bed', sep='\t', header=None, index=None)

```

–ó–∞–ø—É—Å—Ç–∏–º —Å–∫—Ä–∏–ø—Ç:

```bash
$ bash scripts/create_bed.py
```

- *–†–µ–∑—É–ª—å—Ç–∞—Ç*
    
    [hk_genes_ensembl.bed](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/hk_genes_ensembl.bed)
    

–¢–µ–ø–µ—Ä—å –∑–∞–ø—É—Å—Ç–∏–º –≤—ã–ø–æ–ª–Ω–µ–Ω–∏–µ —Ñ—É–Ω–∫—Ü–∏–∏ `geneBody_coverage`, –∏—Å–ø–æ–ª—å–∑—É—è –Ω–∞—à –∫–∞—Å—Ç–æ–º–Ω—ã–π  `.bed` —Ñ–∞–π–ª –¥–ª—è –≥–µ–Ω–æ–≤ –¥–æ–º–∞—à–Ω–µ–≥–æ —Ö–æ–∑—è–π—Å—Ç–≤–∞.

```bash
$ mkdir -p RSeQCresults/geneBody_coverage
$ geneBody_coverage.py -r references/hk_genes_ensembl.bed -i sortedbam -o RSeQCresults/geneBody_coverage/
```

- *–†–µ–∑—É–ª—å—Ç–∞—Ç*
    
    [.geneBodyCoverage.curves.pdf](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/.geneBodyCoverage.curves.pdf)
    

–û—Å—Ç–∞–ª—å–Ω—ã–µ –∫–æ–º–∞–Ω–¥—ã –±—É–¥–µ–º –∑–∞–ø—É—Å–∫–∞—Ç—å c –ø–æ–º–æ—â—å—é —Å–∫—Ä–∏–ø—Ç–∞ –∏—Å–ø–æ–ª—å–∑—É—è `.bed` —Ñ–∞–π–ª –≤—Å–µ–≥–æ –≥–µ–Ω–æ–º–∞.

```bash
#!/bin/bash

# make directories for each tool
mkdir -p RSeQCresults/read_distribution
mkdir -p RSeQCresults/infer_experiment
mkdir -p RSeQCresults/junction_saturation
mkdir -p RSeQCresults/junction_annotation

for infile in $(ls hisatOutput/*/*sorted.bam)
do
  sample=$(basename -a -s _sorted.bam ${infile})
  echo "running read distribution on $sample"
  read_distribution.py  -i $infile -r reference/hg38_RefSeq.bed > RSeQCresults/read_distribution/$sample.readCoverage.txt
  infer_experiment.py  -i $infile -r reference/hg38_RefSeq.bed > RSeQCresults/infer_experiment/$sample.infer_experiment.txt
  junction_saturation.py -i $infile -r reference/hg38_RefSeq.bed -o RSeQCresults/junction_saturation/$sample
  junction_annotation.py -i $infile -r reference/hg38_RefSeq.bed -o RSeQCresults/junction_annotation/$sample
done
```

–ó–∞–ø—É—Å—Ç–∏–º —Å–∫—Ä–∏–ø—Ç

```python
$ bash rseqc_script.sh
```

- –†–µ–∑—É–ª—å—Ç–∞—Ç
    
    –ù–µ–∫–æ—Ç–æ—Ä—ã–µ –ø—Ä–æ—Ñ–∏–ª–∏ –Ω–µ –ø–æ–¥—Ç—è–≥–∏–≤–∞—é—Ç—Å—è –≤ multiqc, –ø–æ—ç—Ç–æ–º—É —è –Ω–∞—Ä–∏—Å–æ–≤–∞–ª –∏—Ö –ø—Ä–∏ –ø–æ–º–æ—â–∏ –ø–∏—Ç–æ–Ω–∞
    
    ![clipping profile](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled%201.png)
    
    clipping profile
    
    ![delition progile](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled%202.png)
    
    delition progile
    
    ![insertion profile](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled%203.png)
    
    insertion profile
    
    ![duplication profile](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled%204.png)
    
    duplication profile
    
    ![mismatch profile](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled%205.png)
    
    mismatch profile
    

–ó–∞–ø—É—Ç–∏–º `multiqc` —á—Ç–æ–±—ã —Å–æ–±—Ä–∞—Ç—å —Ä–µ–∑—É–ª—å—Ç–∞—Ç—ã –≤–º–µ—Å—Ç–µ

```bash
$ multiqc . -f -d -dd 2 --outdir qc --filename multirseqc.html
```

- *–†–µ–∑—É–ª—å—Ç–∞—Ç*
    
    [multirseqc.html](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/multirseqc.html)
    

# –í–∏–∑—É–∞–ª–∏–∑–∞—Ü–∏—è

## Genome Browser

–ó–∞—Ö–æ–¥–∏–º –Ω–∞ —Å–∞–π—Ç [Genome Browser](https://genome.ucsc.edu/cgi-bin/hgCustom), –≤—ã–±–∏—Ä–∞–µ–º –≤ –∫–∞—á–µ—Å—Ç–≤–µ –æ—Ä–≥–∞–Ω–∏–∑–º–∞ –º—ã—à–∫—É –∏ –¥–æ–±–∞–≤–ª—è–µ–º –∫–∞—Å—Ç–æ–º–Ω—ã–π —Ç—Ä–µ–∫. –ü–æ –∫–ª—é—á—É `bigDataUrl` –º—ã –ø–µ—Ä–µ–¥–∞—ë–º —Å—Å—ã–ª–∫—É –Ω–∞ `.bam` —Ñ–∞–π–ª, –ø–æ –∫–ª—é—á—É `bigDataIndex` —Å—Å—ã–ª–∫—É –Ω–∞ `.bai` —Ñ–∞–π–ª. –§–∞–π–ª—ã `.bam` –∏ `.bai` —è –ø—Ä–µ–¥–≤–∞—Ä–∏—Ç–µ–ª—å–Ω–æ –∑–∞–≥—Ä—É–∑–∏–ª –Ω–∞ [figshare](https://figshare.com/articles/dataset/Untitled_Item/25481203?file=45279691).

```
track type=bam name="SRR3414630" bigDataUrl=https://figshare.com/ndownloader/files/45279703 bigDataIndex=https://figshare.com/ndownloader/files/45279691
```

- –í—ã–≥–ª—è–¥–∏—Ç –æ–Ω–æ –∫–∞–∫-—Ç–æ —Ç–∞–∫
    
    ![Untitled](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled%206.png)
    

–ü–æ–ª—É—á–∏–ª–æ—Å—å —á—Ç–æ-—Ç–æ —Ç–∞–∫–æ–µ. –ù–æ –æ—Ç–∫—Ä—ã–≤–∞–µ—Ç—Å—è –Ω–µ –¥–ª—è –≤—Å–µ—Ö —Ö—Ä–æ–º–æ—Å–æ–º (–≤–∏–¥–∏–º–æ –ø–æ—Ç–æ–º—É —á—Ç–æ –Ω—É–∂–Ω–æ –±—ã–ª–æ –≤—ã—Ä–∞–≤–Ω–∏–≤–∞—Ç—å –Ω–∞ –≥–µ–Ω–æ–º –æ—Ç ucsc).

![Untitled](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled%207.png)

## Persephone

–Ø –∑–∞–≥—Ä—É–∑–∏–ª genome.fa, genome.gtf –∏ —Å–≤–æ–π .bam  —Ñ–∞–π–ª –Ω–∞ —Å–∞–π—Ç–µ [Persephone](https://web.persephonesoft.com/). –ò –ø–æ–ª—É—á–∏–ª–æ—Å—å —á—Ç–æ-—Ç–æ —Ç–∞–∫–æ–µ. –í—Å—ë –¥–µ–ª–∞–µ—Ç—Å—è –æ—á–µ–Ω—å –ø—Ä–æ—Å—Ç–æ –∏ –ª–µ–≥–∫–æ, —Å–æ–≤–µ—Ç—É—é.

![Untitled](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled%208.png)

![Untitled](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/Untitled%209.png)

# –ö–≤–∞–Ω—Ç–∏—Ñ–∏–∫–∞—Ü–∏—è

–£—Å—Ç–∞–Ω–æ–≤–∏–º –ø–∞–∫–µ—Ç –∏—Å–ø–æ–ª—å–∑—É—è –∫–æ–Ω–¥—É. –î–ª—è —ç—Ç–æ–≥–æ —Å–æ–∑–¥–∞–¥–∏–º –Ω–æ–≤–æ–µ –æ–∫—Ä—É–∂–µ–Ω–∏–µ –¥–ª—è –Ω–µ–≥–æ

```bash
$ conda create -n htseq
$ conda acivate htseq
$ conda install HTSeq
```

–ù–∞–ø–∏—à–µ–º —Å–∫—Ä–∏–ø—Ç –¥–ª—è –ø–æ–¥—Å—á—ë—Ç–∞ —ç–∫—Å–ø—Ä–µ—Å—Å–∏–∏ –ø—Ä–∏ –ø–æ–º–æ—â–∏ `htseq-count`

```bash
#!/bin/bash
mkdir -p countsHtseq

for infile in sortedbam/*.bam
do
  sample=$(basename -a -s _sorted.bam ${infile})
   htseq-count --format bam \
   --stranded no \
   -i gene_id \
   --counts_outpu countsHtseq/${sample}_counts.tsv \
   $infile \
   references/genome.gtf
done
```

–ó–∞–ø—É—Å—Ç–∏–º —Å–∫—Ä–∏–ø—Ç

```python
$ bash scripts/htseq_quantify.sh
```

–ó–∞–ø—É—Ç–∏–º `multiqc` —á—Ç–æ–±—ã —Å–æ–±—Ä–∞—Ç—å —Ä–µ–∑—É–ª—å—Ç–∞—Ç—ã –≤–º–µ—Å—Ç–µ

```bash
$ multiqc . -f -d -dd 2 --outdir qc --filename multiqcfinal.html
```

- *–†–µ–∑—É–ª—å—Ç–∞—Ç*
    
    [multiqcfinal.html](RNA-seq%202cabfaf5caa9455f98df9839ec0ceb01/multiqcfinal.html)
    

# –î–∏—Ñ—Ñ–µ—Ä–µ–Ω—Ü–∏–∞–ª—å–Ω–∞—è —ç–∫—Å–ø—Ä–µ—Å—Å–∏—è

## –ù–æ—Ä–º–∞–ª–∏–∑–∞—Ü–∏—è

## –°—Ä–∞–≤–Ω–µ–Ω–∏–µ —ç–∫—Å–ø—Ä–µ—Å—Å–∏–π